sdk/core/Azure.Core/src/ResourceType.cs

```diff
@@ -0,0 +1,159 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Globalization;
+using System.Linq;
+
+namespace Azure.Core
+{
+    /// <summary>
+    /// Structure representing a resource type.
+    /// </summary>
+    public readonly struct ResourceType : IEquatable<ResourceType>
+    {
+        /// <summary>
+        /// The resource type for the root of the resource hierarchy.
+        /// </summary>
+        public static readonly ResourceType Root = new ResourceType(string.Empty, string.Empty);
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ResourceType"/> class.
+        /// </summary>
+        /// <param name="resourceType"> The resource type string to convert. </param>
+        public ResourceType(string resourceType)
```

Comment:
How many of these are we creating in common usecases? I'm not a fan of how expensive this ctor is.

Reply:
almost every API call from a service will have a resourcetype string that will be added to this ctor during deserialization